### PR TITLE
fix(scully): Remove exit process before finishing execution

### PR DESCRIPTION
**Pull request by lucasvieirasilva**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/scullyio/scully/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Other... Please describe: Rollback this PR https://github.com/scullyio/scully/pull/1334 changes.

## What is the current behavior?

Currently, the `process.exit(0)` in the `startup.ts` is making these code https://github.com/scullyio/scully/blob/main/libs/scully/src/scully.ts#L164-L171 never be executed, leaving the `serve` process opened in the port `1864`

Issue Number: https://github.com/scullyio/scully/issues/1397

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

To Scully runs without any problems in the CI/CD environments, this issue should be resolved first: https://github.com/scullyio/scully/issues/1348

### CODE REVIEW

**Comment by SanderElias** [comment]
@lucasvieirasilva Thanks for the pr. But the hard exit is there for e reason. It is a fix for the occasional stall in CI/CD.
We are still looking for the root cause of that, but without the hard exit, this happens even more. 
Also, we are experimenting with different way's to start the background server, and take better control over it, so we can indeed properly it the thing.

**Comment by SanderElias** [comment]
Closing because this issue is under investigation, and will be solved later on

**Comment by lucasvieirasilva** [comment]
> Closing because this issue is under investigation, and will be solved later on

@SanderElias, the hard exit introduced in PR #1334 is not a 'fix' for the stalling in CI/CD, but just exits Scully before it tries to close the background webserver, which is where the real CI stalling problem is. A consequence of this is that the web server is now always left running, so any subsequent executions of Scully within the same CI will use the previous web server execution. This is an issue when a different Scully configuration is required for the subsequent execution. This is documented in Issue #1397.
So in summary, PR #1334 has just 'avoided' one issue but caused another, and should be backed out.
